### PR TITLE
[License] update license SPDX in all package.json, source file headers

### LINKED ISSUE #12583: [License] Inaccurate mentions of secondary license name and SPDX
### Bug Description:

@SpessartZiegler recently [reported a couple problems with our licenses](https://github.com/eclipse-theia/theia/pull/12424#issuecomment-1551349506):

- ~~We have had the SPDX wrong for our secondary license [1]. Also in places, we have that license name wrong. I checked for the primary license (EPL) and found no similar issues with our naming/referencing it.~~ update: Wayne confirmed that the current SPDX is not perfect but better than the deprecated version. 
- When we recently split the license file into two, the text of the secondary license got accidently changed, e.g. now it contains an extra, [Oracle-specific pargraph](https://github.com/eclipse-theia/theia/blob/e9cd6873e05b00b2c2b2ad7eefa486bf3e8a3a9c/LICENSE-GPL#L343-L347).  We should re-align on the official license text.

In addition, I think that the naming of the `LICENSE-GPL` file is likely to cause confusion and that a name that's more in-line with the fact that the secondary license is specifically the "classpath exception" variant, would be an improvement.

Some more details: 

In the `package.json` files and in the header of source files, we have the SPDX\[1\], and a slightly better one suggested by Wayne :
- current: `SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0`
- better:  `SPDX-License-Identifier: EPL-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0`

Links:
- \[1\]: https://spdx.org/licenses/GPL-2.0-with-classpath-exception.html
- \[2\]: https://spdx.org/licenses/EPL-2.0.html

### Steps to Reproduce:

1.
2.
3.

### Additional Information

- Operating System:
- Theia Version:

### PR DESCRIPTION
**Pull request by marcdumais-work**
#### What it does

_update: following Wayne's feedback below, I have updated the description and content of this PR_

This PR tweaks the secondary license SPDX short identifier, to use the most precise version that applies. It also restores the original text of secondary license, that was accidentally changed during the recent file split. 

Making these changes should make our packages easier to deal-with, from an IP PoV, for adopters and extenders, by making our licenses easier to auto-detect. 

For more details see the [corresponding issue](#12583 )

Note: ~~Since this PR changes our copyright headers and the stated license in various `package.json` files, I have created an IP ticket, for the Foundation IP Team to sign-off on these changes: [ticket](https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/8699). We should probably wait (timeout: 3 weeks?) for feedback before merging.~~

The feedback was received and acted-upon. This PR updates the license headers to use the more precise, suggested SPDX:

Before:
`SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0`

After:
`SPDX-License-Identifier: EPL-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0`

Fixes #12583

#### How to test

This PR makes changes to almost 2000 files. As such I have experienced slow load/update times, working in the GitHub PR diff view (Files Changed tab). It might be preferable to review the changes locally. I have split things in separate commits, so that important changes do not go unnoticed, among the mass of license header changes. Please consider looking at the PR commit-by-commit. We can squash into a single commit at the end, before merging.

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,16 +19,16 @@ Eclipse Theia is an extensible framework to develop full-fledged multi-language
 
 </div>
 
-- [**Website**](#website)
```
Comment (marcdumais-work) [inline README.md:22]: This was not an intended change. When I try to change it back locally, the stars `**` are automatically removed. I am guessing that using them is against some config we have somewhere. I think I would just leave out these stars, unless there are objections.

**Review by vince-fugnitto — APPROVED** [review]
The changes look good to me :+1: 
- the splitting looks correct
- the new `header` snippet works as intended
- the updated eslint rule correctly identifies errors with the header if missing, or mistyped

**Comment by marcdumais-work** [comment]
> I believe that GPL-2.0-only is correct for Eclipse Theia.

Agreed.

> The current form is actually more correct than the changes in this PR.

> Note that while changing GPL-2.0 to GPL-2.0-only will make the headers more correct, most SPDX parsers will still be able to make sense of GPL-2.0. That is, the EMO is supportive of this change but considers it low priority.

Understood, thanks @waynebeaton!

**Review by vince-fugnitto — APPROVED** [review]
I re-affirm my previous review :+1:

**Comment by marcdumais-work** [comment]
@msujew I wanted to mention that I tested the GitHub detection of our licenses, using my fork (the only consistent way I found was to merge to master and observe how it affected the detection). Doing so, I have confirmed that even with tweaked SPDX, the licenses were not detected correctly when I tried going-back to a single license file. So, I think your split of the single license file, into two files, was the correct move!

**Comment by marcdumais-work** [comment]
Thanks for the review @vince-fugnitto . I will squash the commits locally and tidy-up the commit message.

**Comment by marcdumais-work** [comment]
I have just rebased this PR locally to confirm that linting still passes (in case a PR had been merged in the meantime, that used the old SPDX in source header. So, there should be no surprises when we merge. It's possible that there are PRs under review that will need to update new headers - the safest will be to rebase on latest master once this one here is in.

**Review by msujew — APPROVED** [review]
Thank you for the work on this @marcdumais-work, looks good to me as well 👍

**Pull request merged**